tests/foreman/api/test_host.py

```diff
@@ -1572,149 +1321,70 @@ def test_positive_read_puppet_ca_proxy_name(self):
-                interface.name = new_name
-                with self.assertRaises(HTTPError) as error:
-                    interface.update(['name'])
-                self.assertNotEqual(interface.read().name, new_name)
-                self.assertEqual(error.exception.response.status_code, 422)
-
-    @tier1
-    def test_positive_delete(self):
-        """Delete host's interface (not primary)
-
-        :id: 9bf83c3a-a4dc-420e-8d47-8572e5ae1dd6
-
-        :expectedresults: An interface is successfully deleted
-
-        :CaseImportance: Critical
-        """
-        host = entities.Host().create()
-        interface = entities.Interface(host=host).create()
+        name = gen_choice(valid_interfaces_list())
+        interface = entities.Interface(host=module_host, name=name).create()
+        assert interface.name == name
+        new_name = gen_choice(valid_interfaces_list())
+        interface.name = new_name
+        interface = interface.update(['name'])
+        assert interface.name == new_name
         interface.delete()
-        with self.assertRaises(HTTPError):
+        with pytest.raises(HTTPError):
             interface.read()
 
     @tier1
-    def test_negative_delete(self):
-        """Attempt to delete host's primary interface
+    def test_negative_end_to_end(self, module_host):
+        """Attempt to create and update an interface with different invalid entries as names
+        (>255 chars, unsupported string types), at the end attempt to remove primary interface
 
-        :id: 716a9dfd-0f31-45aa-a6d1-42add032a15c
+        :id: 6fae26d8-8f62-41ba-a1cc-0185137ef70f
 
-        :expectedresults: An interface is not deleted
+        :expectedresults: An interface is not created, not updated
+            and primary interface is not deleted
 
         :CaseImportance: Critical
         """
-        host = entities.Host().create()
+        name = gen_choice(invalid_interfaces_list())
+        with pytest.raises(HTTPError) as error:
+            entities.Interface(host=module_host, name=name).create()
+            assert error.exception.response.status_code == 422
+        interface = entities.Interface(host=module_host).create()
+        interface.name = name
+        with pytest.raises(HTTPError) as error:
+            interface.update(['name'])
+            assert interface.read().name != name
+            assert error.exception.response.status_code == 422
+
         primary_interface = next(
-            interface for interface in host.interface if interface.read().primary
+            interface for interface in module_host.interface if interface.read().primary
         )
-        with self.assertRaises(HTTPError):
+        with pytest.raises(HTTPError):
             primary_interface.delete()
-        with self.assertNotRaises(HTTPError, expected_value=404):
+        try:
             primary_interface.read()
+        except HTTPError:
+            pytest.fail("HTTPError raised value 404 unexpectedly!")
```

Comment:
"HTTPError raised value 404 unexpectedly!" is a very weird wording. I suggest "HTTPError 404 raised unexpectedly" instead.

Reply:
DONE